canonical-web-and-design/jaas-dashboard
# Add checkbox column to the unit list.

## Done

- Convert App.js to TypeScript.
- Convert tableHeaders.js to TypeScript.
- Clean up invalid call signature for `generateIdentityIdentifier`.
- Fix some types in ButtonGroup.tsx, EntityInfo.tsx.
- Created generic `FormikFormData` component to extract form change events from Formik forms.
- Converted the table rendering to be more fluid. The table column sizing fixes will come with the `ModularTable` update.
- Add checkbox to each unit row only in the App view.
- Add select-all checkbox to the header for the unit table only in the App view.

## QA

- Ask me for model access to `action-test` if you don't already have.
- Select the Select-All checkbox, it should select them all.
- De-select the Select-All checkbox, it should de-select them all.
- Select then de-select a single unit row.
- Select each units checkbox individually, once all have been selected, the select all checkbox should select
- De-select one of the units, the select-all checkbox should de-select
- Select all units again and then de-select the select-all, it should de-select all the units.

## Details

Fixes: https://github.com/canonical-web-and-design/juju-squad/issues/1654

## Screenshots

![Screen Shot 2021-03-31 at 3 05 06 PM](https://user-images.githubusercontent.com/532033/113211265-8017de00-9232-11eb-9579-89a7a6c7ca18.png)
